conbench/app/_plots.py

```diff
@@ -237,6 +238,28 @@ def time_series_plot(history, benchmark, run, height=250, width=1000):
         legend_label="Benchmark",
         name="benchmark",
     )
+
+    step_changes = [
+        Span(
+            location=dateutil.parser.isoparse(result["timestamp"]),
+            dimension="height",
+            line_color="black",
+        )
+        for result in history
+        if result["is_step_change"]
+    ]
+    for step_change in step_changes:
+        p.add_layout(step_change)
+    if step_changes:
+        # hack: add a dummy line so it appears on the legend
+        one_result = [result for result in history if result["is_step_change"]][0]
+        p.line(
+            [dateutil.parser.isoparse(one_result["timestamp"])] * 2,
+            [one_result["mean"]] * 2,
+            legend_label="Step change",
+            line_color="black",
```

Comment:
I think black is visually confusing here, since it can look like part of the line itself (i.e. a very fast vertical wiggle, which it kind of is, but not like that). Red comes to mind, but that's more appropriate for regressions than improvements; if we're not going to separate them, we should probably use a more neutral color like blue or purple or orange or something. Or we could color based on direction if we track it (which it seems like we should). We may need special handling for spikes that immediately revert then though.

Opacity would help too; maybe 50-70% so it's more of an event than the distribution ribbon, but less of a datapoint than the line.

Reply:
These are great ideas. Maybe a dotted line would work well.

Reply:
Agreed here, that the black might be a bit stark. I suspect we will need to treat the two cases we are currently imagining slightly differently (especially later in the history after more runs are added to the plots):

* if it's a one-off sudden regression or improvement that goes back to the mean right after  
    here, we probably want this point to be lowered opacity to show that it's being ignored 
    
* if it's a step-change and the benchmarks are now at this new higher or lower number  
    here, we probably want to highlight where this began, and what all is included in determining variability here.

Reply:
Yes, agreed. I'm planning to handle the case of "spikes" in #474 . Those will be rendered differently on the plots. 🙂 

I'll attach a screenshot of the new line style.

Reply:
Here's the new line style (and the new button below the plot) 
<img width="1149" alt="Screenshot 2023-01-06 at 14 38 02" src="https://user-images.githubusercontent.com/16600275/211096236-462d082f-55a6-408a-b73f-4328c4187aaf.png">

Reply:
Update: 
<img width="1238" alt="Screenshot 2023-01-10 at 16 07 42" src="https://user-images.githubusercontent.com/16600275/211673346-d5b8cff8-3df3-4f91-9582-2fa6c66f521f.png">